File: program (103).java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//  Function Name  :    Reverse
//  Description :       Accept String from user and display the String in reverse. 
//  Input :             String 
//  Output :            String
//  Author :            Aman Shaikh    
//  Date:               14/12/2024 
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class StringDemo
{
    public void Reverse(String str)
    {
        StringBuilder s2 = new StringBuilder();
        int i = str.length() - 1;
        while(i >= 0) 
        {
            s2.append(str.charAt(i));
            i--;
        }
        System.out.println("Reversed String: " + s2.toString());
    }
}

class Assignment36E 
{
    public static void main(String Arr[])
    {
        Scanner ssobj = new Scanner(System.in);
        System.out.println("Enter the String : ");
        String s1 = ssobj.nextLine(); 

        StringDemo sobj = new StringDemo();
        sobj.Reverse(s1); 
    }
}

File: program (103).cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//  Function Name  :    Mult                                        
//  Description :       Accept number from user and return product of its digit using Recursion.
//  Input :             int 
//  Output :            int
//  Author :            Aman Shaikh    
//  Date:               15/12/2024 
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

class Pattern
{
    public:
    int Mult(int iNo) 
    {
        int iNum = 0;
        static int iMult = 1; 

        if(iNo != 0) 
        {
            iNum = iNo % 10;
            if(iNum != 0)
            {
                iMult = iMult * iNum;
            }
            iNo = iNo / 10;
            Mult(iNo);
        } 
        return iMult;
    }
};

int main()
{
    int iValue = 0, iRet = 0;

    cout<<"Enter the number : "; 
    cin>>iValue;

    Pattern pobj;
    iRet = pobj.Mult(iValue);

    cout<<"Product is : "<<iRet<<"\n";

    return 0;
}
